delfin/drivers/dell_emc/vnx_block/component_handler.py

```diff
@@ -0,0 +1,477 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import six
+from oslo_log import log
+from oslo_utils import units
+
+from delfin import exception
+from delfin.common import constants
+
+LOG = log.getLogger(__name__)
+
+
+class ComponentHandler():
+
+    EMCVNX_VENDOR = 'DELL EMC'
+    RAID_GROUP_ID_PREFIX = 'raid_group_'
+
+    STATUS_MAP = {'Ready': constants.StoragePoolStatus.NORMAL,
+                  'Offline': constants.StoragePoolStatus.OFFLINE,
+                  'Valid_luns': constants.StoragePoolStatus.NORMAL,
+                  'Busy': constants.StoragePoolStatus.ABNORMAL,
+                  'Halted': constants.StoragePoolStatus.ABNORMAL,
+                  'Defragmenting': constants.StoragePoolStatus.NORMAL,
+                  'Expanding': constants.StoragePoolStatus.NORMAL,
+                  'Explicit Remove': constants.StoragePoolStatus.OFFLINE,
+                  'Invalid': constants.StoragePoolStatus.OFFLINE,
+                  'Bound': constants.StoragePoolStatus.NORMAL}
+
+    VOL_TYPE_MAP = {'no': constants.VolumeType.THICK,
+                    'yes': constants.VolumeType.THIN}
+
+    VOL_COMPRESSED_MAP = {'no': False,
+                          'yes': True}
+
+    def __init__(self, navi_handler=None):
+        self.navi_handler = navi_handler
+
+    def get_storage(self):
+        try:
+            storage = self.navi_handler.get_agent()
+            status = constants.StorageStatus.NORMAL
+
+            used_cap = 0
+            free_cap = 0
+            raw_cap = 0
+            try:
+                free_map = self.handler_storage_free_capacity()
+                if free_map:
+                    raw_cap = free_map.get('raw_cap')
+                    free_cap = free_map.get('free_cap')
+                    if free_cap == 0:
+                        status = constants.StorageStatus.ABNORMAL
+            except Exception:
+                LOG.error('Get storage free capacity and raw capacity failed!')
+
+            try:
+                used_map = self.handler_storage_used_capacity()
```

Comment:
Hope we can combine  .handler_storage_free_capacity and  .handler_storage_used_capacity to one function, both are handling Pool capacity